pysph/base/tests/test_opencl.py

```diff
@@ -4,8 +4,55 @@
 
 pytest.importorskip("pysph.base.opencl")
 
+import pyopencl as cl
+
 from pysph.base.utils import get_particle_array  # noqa: E402
-from pysph.base.opencl import DeviceHelper  # noqa: E402
+from pysph.base.opencl import DeviceHelper, DeviceArray, \
+        get_queue  # noqa: E402
+
+
+class TestDeviceArray(TestCase):
+    def test_reserve(self):
+        # Given
+        self.dev_array = DeviceArray(np.int32, n=16)
+        self.dev_array.fill(0)
+        self.dev_array.array[0] = 1
+        # When
+        self.dev_array.reserve(64)
+        # Then
+        assert len(self.dev_array.get_data()) == 64
+        assert self.dev_array.length == 16
+        assert self.dev_array.array[0] == 1
+
+    def test_resize_with_reallocation(self):
+        # Given
+        self.dev_array = DeviceArray(np.int32)
+        # When
+        self.dev_array.resize(64)
+        # Then
+        assert len(self.dev_array.get_data()) == 64
+        assert self.dev_array.length == 64
```

Comment:
You should also test that the content of the old data is preserved after resize.